Automattic/wp-calypso
# Upgrade to core-js@3.

It's starting to take a toll to still use `core-js@2`, since we're missing important features: `Object.fromEntries`, `Array.flat`, `globalThis`, `URL`, to name a few. We've been adding support for some of these with individual polyfills, but it's time to take the leap and move over to `core-js@3`, to ensure our development environment stays representative of the current state of the JS spec.

This will initially result in some code duplication, as unfortunately we use several packages with a hardcoded dependency on `core-js@2`. Hopefully, over time, we'll be able to replace these packages with more modern ones, or update them ourselves to be compatible with `core-js@3`.

#### Changes proposed in this Pull Request

* Update the `core-js` package to the current version.
* Set the `core-js` version in the babel config.
* Drop `@babel/polyfill` and use `core-js` directly.
* Remove superfluous polyfills (e.g. `@webcomponents/url`).
* Remove superfluous polyfill module replacements in evergreen.
* Modify server entry point to use `boot/polyfills` rather than its own code.
* Minor lint fixes.

#### Testing instructions

This change affects all of Calypso, so please be sure to try using it as much as possible both in newer browsers, which should use the evergreen build (that includes fewer polyfills); and in older browsers (particularly IE11), which should use the fallback build (that includes all polyfills).

Potential areas of interest: anything that does URL processing (e.g. sharing from Reader), and anything that uses `globalThis` (e.g. language switching).
